hydra-main/src/main/java/com/addthis/hydra/minion/LogUtils.java

```diff
@@ -93,8 +94,16 @@ public static JSONObject readLogLines(File file, int startOffset, int lines) {
                     }
                 }
                 bytesRead = (int) (len - off);
-                byte[] buf = new byte[bytesRead];
-                raf.read(buf);
+                byte[] buf = null;
+                // limiting log reads below 20MB, in case of reaching heap limit and crashing minion
+                if(bytesRead >= logBufLimit) {
```

Comment:
Each of these 4 sections could be replaced by a call to a new method you can create. the raf.seek() calls can be made before the call to the method. Something like "readFileBytesLimited(long requestedBytes)" which returns a new byte[] created in the method.